src/main/java/com/aws/greengrass/config/Topics.java

```diff
@@ -380,6 +380,29 @@ public boolean equals(Object o) {
         return false;
     }
 
+    /**
+     * Checks if the Topics has the same name and children.
+     *
+     * @param t the Topics to compare
+     */
+    public boolean compareChildren(Topics t) {
+        if (t == null || !getName().equals(t.getName()) || children.size() != t.children.size()) {
+            return false;
+        }
+        for (Map.Entry<CaseInsensitiveString, Node> me : children.entrySet()) {
+            Object themObject = t.children.get(me.getKey());
+            Object meObject = me.getValue();
+            if (meObject instanceof Topics && themObject instanceof Topics) {
+                return ((Topics)meObject).compareChildren((Topics) themObject);
```

Comment:
you cannot just return here. If this is true, but another child is false then the wrong answered is returned